app/Main.hs

```diff
@@ -171,6 +198,24 @@ optsParser =
       [ metavar "FILE",
         help "Haskell source files to format or stdin (the default)"
       ]
+    <*> cabalDefaultExtensionsParser
+
+cabalDefaultExtensionsParser :: Parser CabalDefaultExtensionsOpts
+cabalDefaultExtensionsParser =
+  CabalDefaultExtensionsOpts
+    <$> (switch . mconcat)
+      [ short 'e',
+        long "cabal-default-extensions",
+        help "Account for default-extensions from .cabal files"
+      ]
+    <*> (optional . strOption . mconcat)
```

Comment:
> If `--cabal-file` is not given yet `--cabal-default-extensions` is passed error out and complain about the missing `--cabal-file`, i.e. it becomes obligatory in that case. We could define a new `OrmoluException` for this, with exit code 9.

A problem with this is that editor integrations have to reimplement locating the .cabal file.

---

Another possibility: Just remove `--cabal-file`. I actually can't think of any situation without a trivial workaround where the current logic of `findCabalFile` would fail.

Reply:
Let's remove `--cabal-file` then.

Reply:
But we still need a new Ormolu exception because there will be cases when `--cabal-default-extensions` is passed but no `--stdin-input-file`. We should complain in those cases.